The reason different programs exist is that they offer **tradeoffs in speed and accuracy**.

We'll explore how the **same set of globins** can be **aligned differently** with various programs and try to asses which alignments are most accurate.

# Consequences of misalignment

- **Conservation of critial residues** (e.g. active site in enzyme, heme-binding residues...) may be missed.
- **Phylogenetic interference** may be compromised (all depend on [[Introduction to Multiple Sequence Alignment|MSA]] as input).
- **Protein structure prediction** is compromised by faulty MSA (its often a first step in homology-based modeling).

# Different approaches to MSA

Ultimately there are two main groups of programs, based on the accuracy of if the alignments are optimal.

1. [[Exact approaches to MSA|Exact approaches]]
2. [[Non-exact approaches to MSA|Heuristic approaches]]

